packages/core/src/getStateFromPath.tsx

```diff
@@ -289,23 +290,34 @@ const matchAgainstConfigs = (remaining: string, configs: RouteConfig[]) => {
     if (match) {
       const matchedParams = config.pattern
         ?.split('/')
-        .filter((p) => p.startsWith(':'))
+        .map((p, x) => [p, x] as const)
+        .filter(([p]) => p.startsWith(':'))
         .reduce<Record<string, any>>(
-          (acc, p, i) =>
+          (acc, [p, x], i) =>
             Object.assign(acc, {
               // The param segments appear every second item starting from 2 in the regex match result
-              [p]: match![(i + 1) * 2].replace(/\//, ''),
+              // Also include position in the key to handle the same param appearing multiple times
+              [p + x]: match![(i + 1) * 2].replace(/\//, ''),
             }),
           {}
         );
 
       routes = config.routeNames.map((name) => {
         const config = configs.find((c) => c.screen === name);
-        const params = config?.path
+
+        // Get the number of segments in the initial pattern
+        const initialPartsLength = config?.pattern
+          .replace(new RegExp(`${escape(config.normalizedPath)}$`), '')
+          ?.split('/').length;
+
+        const params = config?.normalizedPath
           ?.split('/')
-          .filter((p) => p.startsWith(':'))
-          .reduce<Record<string, any>>((acc, p) => {
-            const value = matchedParams[p];
+          .map((p, x) => [p, x] as const)
+          .filter(([p]) => p.startsWith(':'))
+          .reduce<Record<string, any>>((acc, [p, x]) => {
```

Comment:
Don't use `any`

Reply:
Got it! I'll update you use unknown instead if that's okay.